app/components/post_draft/post_draft.js

```diff
@@ -89,8 +94,13 @@ export default class PostDraft extends PureComponent {
     }
 
     componentDidMount(prevProps) {
+        const {useGroupMentions} = this.props;
+
         if (this.props.isTimezoneEnabled !== prevProps?.isTimezoneEnabled || prevProps?.channelId !== this.props.channelId) {
             this.numberOfTimezones().then((channelTimezoneCount) => this.setState({channelTimezoneCount}));
+            if (useGroupMentions) {
+                this.props.getChannelMemberCountsByGroup(this.props.channelId, this.props.isTimezoneEnabled);
```

Comment:
so when this component mounts, you are dispatching this action, which as I said above will the cause a re-render to this component, the problem is that this component will mount for the channel screen  and every thread screen you open, thus there will be a re-render for this component in the channel screen and in the thread screen, and they pile up, multiple unnecessary re-renders translates in performance degradation.

Also not sure how in the hell this happened but `componentDidMount` does not have `prevProps` or any argument whatsoever, so I guess that was a bad merge, it should all be moved to **componentDidUpdate**

Reply:
Moved to `componentDidUpdate`. Also, there is a check to call this action if either the `isTimezoneEnabled` has changed or `channel` has changed. I checked and when I switch threads, this action is not being called (I was console logging in `RECEIVED_CHANNEL_MEMBER_COUNTS_BY_GROUP` reducer and did not notice any console logging when click on threads. Only when I switched channels and only if those channels had group members in them did the I get a action dispatch.